### DISCUSSION TITLE
Spatial calibration - ImageToProbeTransform

### DISCUSSION THREAD

**Discussion by NesBarn** [PlusToolkit/PlusLib#885] [Q&A]
Hello,

For automatic spatial calibration, I used the application fCal. I went through the different steps and I got this following image to probe transform matrix:
[-0.0381 -0.034 0.1459 -21.773
0.165 -0.0023 0.0333 -60.9938
0.0018 0.1348 0.0374 104.6644
0 0 0 1].
Normally I should obtain an orthogonal matrix (MM-1=I) but it is not the case with this one.
Can you help me please regarding this matrix?
Thank you

**Comment by lassoan** [comment]
When the resulting transform is non-orthogonal then it indicates that there is an inconsistency (inaccurate tracking, pivot calibration, phantom registration, low image quality, etc.). You can enforce orthogonality constraint by specifying "OptimizationMethod". If you also know that the spacing is isotropic (it is, if the actual speed of sound in the material is the same what is set in the ultrasound imaging device) then you can set `IsotropicPixelSpacing="TRUE"`.

For example
```
  <vtkPlusProbeCalibrationAlgo 
    ImageCoordinateFrame="Image" 
    ProbeCoordinateFrame="Probe" 
    PhantomCoordinateFrame="Phantom" 
    ReferenceCoordinateFrame="Reference" 
    OptimizationMethod="2D" 
    IsotropicPixelSpacing="TRUE" />
```

See details here: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/AlgorithmProbeCalibration.html

**Comment by NesBarn** [comment]
Hello,
Thank you for your answer. I tried as suggested. The problem is that I always get the message "Calibration error is too high" even when the probe (on which is mounted the tracker tool) is fixed and doesn't move... I don't know from where comes the problem (error of 0.41mm during Phantom registration and 0.35 mm during pivot calibration). For information, if it may help, I needed to set "PortUsImageOrientation="UF" in order to obtain the right pattern of the wires (see image attached) (because if the probe is fixed with its marker in front of the "M" over the phantom and the "PortUsImageOrientation" set to "UF", I have the wrong pattern (wires flipped)
![image](https://user-images.githubusercontent.com/97302106/148777748-997d84b4-fe8c-4671-a717-e22df1d85485.png)
![image](https://user-images.githubusercontent.com/97302106/148777783-45ffb3f3-1948-423c-a9ca-6c1c0ec4ed11.png)

**Comment by jonortega77** [comment]
any update regarding this issue?
